utils/schema.py

```diff
@@ -23,7 +24,19 @@ class Meta:
     )
 
     def resolve_count(self, info):
-        return self.iterable.distinct().count()
+        if isinstance(self.iterable, QuerySet):
+            return self.iterable.distinct().count()
+
+        return len(set(self.iterable))
 
     def resolve_total_count(self, info, **kwargs):
-        return self.iterable.model.objects.count()
+        if isinstance(self.iterable, QuerySet):
+            return self.iterable.model.objects.count()
+
+        # Because the DataLoader patter returns only a list of items rather than a QuerySet,
+        # it's not possible to get the model from the iterable, so we try to get the model based
+        # on the connection node.
+        if model := getattr(self._meta.node._meta, "model", None):
+            return model.objects.count()
+
+        return len(self.iterable)
```

Comment:
> When does this happen?

When the response is taken from the DataLoader rather than the django QuerySet that is normally returned from this connection fields.

> Is it possible that DataLoader can be made to return a subclass of list with extra "model" attribute?

It's not that straightforward. Since the `FiltersetClass` takes several attributes from the queryset itself, it's not as straightforward as just adding the "missing" attributes.